### ISSUE TITLE
Selenium::WebDriver::Error::WebDriverError: unexpected response, code=200 content-type=""

### ISSUE THREAD

**Issue by lvonk** [SeleniumHQ/selenium#9130]
## 🐛 Bug Report

We are upgrading our project to ruby 3.0 and receive the following error when doing the necessary updates of all gems including the `selenium-webdriver`  (see entire stacktrace below): 

`Selenium::WebDriver::Error::WebDriverError:  unexpected response, code=200 content-type=""`

The setup is as follows:

- ruby-3.0
- docker image `selenium/standalone-chrome:4.0.0`
- selenium-webdriver (~> 4.0.0.alpha7)
- webdrivers (4.5.0)
- capybara (3.35.3)

Behavior is similar in running on Mac OSX and on e.g. circleci with linux images

I tracked it down to `Selenium::WebDriver::Remote::Http::Common.create_response`. It receives an empty `content_type`. When monkeypatching (🙈 ) this method and checking if the content looks like json `if content_type.include?('application/json') || body.start_with?('{')` it works. Of course this is not a proper solution but it shows that some component (I guess the `selenium/standalone-chrome:4.0.0`) is not returning the `content_type` correctly for retrieving the current browser.

## To Reproduce

Create a project with the above gems and call `browser = Capybara.current_session.driver.browser`

## Expected behavior

It returns a correct content_type or knows how to handle an otherwise valid response.

## Environment

OS: OSX, Ubuntu
Browser: Chrome
Browser version: '87.0.4280.88'
Browser Driver version: 
Language Bindings version: 
Selenium Grid version (if applicable):

## Entire stacktrace
```ruby
            browser = Capybara.current_session.driver.browser
            
            Selenium::WebDriver::Error::WebDriverError:
              unexpected response, code=200, content-type=""
              {
                "value": {
                  "sessionId": "d0de119d9869a46b7aa672b39051caa1",
                  "capabilities": {
                    "acceptInsecureCerts": false,
                    "browserName": "chrome",
                    "browserVersion": "88.0.4324.96",
                    "chrome": {
                      "chromedriverVersion": "88.0.4324.96 (68dba2d8a0b149a1d3afac56fa74648032bcf46b-refs\u002fbranch-heads\u002f4324@{#1784})",
                      "userDataDir": "\u002ftmp\u002f.com.google.Chrome.1byTAz"
                    },
                    "goog:chromeOptions": {
                      "debuggerAddress": "localhost:46515"
                    },
                    "networkConnectionEnabled": false,
                    "pageLoadStrategy": "normal",
                    "platformName": "linux",
                    "proxy": {
                    },
                    "se:options": {
                      "cdp": "http:\u002f\u002f192.168.176.3:4444\u002fsession\u002fd0de119d9869a46b7aa672b39051caa1\u002fse\u002fcdp"
                    },
                    "setWindowRect": true,
                    "strictFileInteractability": false,
                    "timeouts": {
                      "implicit": 0,
                      "pageLoad": 300000,
                      "script": 30000
                    },
                    "unhandledPromptBehavior": "dismiss and notify",
                    "webauthn:extension:largeBlob": true,
                    "webauthn:virtualAuthenticators": true
                  }
                }
              }
            # ./vendor/bundle/ruby/3.0.0/gems/selenium-webdriver-4.0.0.alpha7/lib/selenium/webdriver/remote/http/common.rb:98:in `create_response'
            # ./vendor/bundle/ruby/3.0.0/gems/selenium-webdriver-4.0.0.alpha7/lib/selenium/webdriver/remote/http/default.rb:110:in `request'
            # ./vendor/bundle/ruby/3.0.0/gems/selenium-webdriver-4.0.0.alpha7/lib/selenium/webdriver/remote/http/common.rb:64:in `call'
            # ./vendor/bundle/ruby/3.0.0/gems/selenium-webdriver-4.0.0.alpha7/lib/selenium/webdriver/remote/bridge.rb:567:in `execute'
            # ./vendor/bundle/ruby/3.0.0/gems/selenium-webdriver-4.0.0.alpha7/lib/selenium/webdriver/remote/bridge.rb:52:in `create_session'
            # ./vendor/bundle/ruby/3.0.0/gems/selenium-webdriver-4.0.0.alpha7/lib/selenium/webdriver/common/driver.rb:328:in `create_bridge'
            # ./vendor/bundle/ruby/3.0.0/gems/selenium-webdriver-4.0.0.alpha7/lib/selenium/webdriver/common/driver.rb:77:in `initialize'
            # ./vendor/bundle/ruby/3.0.0/gems/selenium-webdriver-4.0.0.alpha7/lib/selenium/webdriver/remote/driver.rb:46:in `initialize'
            # ./vendor/bundle/ruby/3.0.0/gems/selenium-webdriver-4.0.0.alpha7/lib/selenium/webdriver/common/driver.rb:61:in `new'
            # ./vendor/bundle/ruby/3.0.0/gems/selenium-webdriver-4.0.0.alpha7/lib/selenium/webdriver/common/driver.rb:61:in `for'
            # ./vendor/bundle/ruby/3.0.0/gems/selenium-webdriver-4.0.0.alpha7/lib/selenium/webdriver.rb:90:in `for'
            # ./vendor/bundle/ruby/3.0.0/gems/capybara-3.35.3/lib/capybara/selenium/driver.rb:83:in `browser'
```

**Comment by bauerd** [comment]
I revived an old-ish project that uses Selenium today and ran into the same problem.
The same problem exists with the standalone Firefox Docker images.
Could you share your monkeypatch, please?

**Comment by lvonk** [comment]
@bauerd 

```ruby
# frozen_string_literal: true

module Selenium
  module WebDriver
    module Remote
      module Http
        module SeleniumPatches
          private

          def create_response(code, body, content_type)
            code = code.to_i
            body = body.to_s.strip
            content_type = content_type.to_s

            WebDriver.logger.info("<- #{body}")

            if content_type.include?('application/json') || body.start_with?('{')
              fail Error::WebDriverError, "empty body: #{content_type.inspect} (#{code})\n#{body}" if body.empty?

              Response.new(code, JSON.parse(body))
            elsif code == 204
              Response.new(code)
            else
              msg = if body.empty?
                      "unexpected response, code=#{code}, content-type=#{content_type.inspect}"
                    else
                      "unexpected response, code=#{code}, content-type=#{content_type.inspect}\n#{body}"
                    end

              fail Error::WebDriverError, msg
            end
          end
        end
      end
    end
  end
end

module Selenium
  module WebDriver
    module Remote
      module Http
        # @api private
        class Common
          prepend SeleniumPatches
        end
      end
    end
  end
end
```

The addition of `|| body.start_with?('{')` in the if statement is the patch. It is a poor man's attempt if the body looks like json, but it does the job for us.

**Comment by titusfortner** [comment]
This *shouldn't be an issue with alpha 7

It was broken January 17, when this commit was merged: 7113b470bdfd1b277a589853554a9521eb3c0ea2
and fixed February 3 with 7cae246a9cd41ca9acf1b0abc81dd6ad3d3f2665

@diemol - how does Docker versioning work? Was `selenium/standalone-chrome:4.0.0` built from trunk between 1/17 & 2/3? 

Regardless, the beta will get released "very soon" and have this fix.
